alica_solver_interface/include/alica_solver_interface/Interval.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <algorithm>
+#include <ostream>
+namespace alica
+{
+template <typename T>
+class Interval
+{
+  public:
+    Interval() {}
+    Interval(T min, T max)
+        : _min(min)
+        , _max(max)
+    {
+    }
+
+    Interval<T>(const Interval<T>& o) = default;
+    Interval<T>& operator=(const Interval<T>& o) = default;
+
+    T getMin() const { return _min; }
+    T getMidPoint() const { return (_max + _min) / 2; }
+    T getMax() const { return _max; }
+    void limitTo(T min, T max)
```

Comment:
just a minor thing, but I think that this function name does not express well what the function does, which is to further constrain the interval than it already is. Maybe sth like `constrainLimitTo` or `addLimit` would be more appropriate.

Reply:
It *limits* the interval *to* a certain range, no?

Reply:
Mathematically speaking, this function is the `∩=`operation. I think this could be more clearly expressed than the function name, but yeah you can also leave it as it is.

On that note, could you implement the `intersect` function such that it re-uses this function, e.g. sth like 
```
Interval intersect(const Interval<T> o) const {Interval ret(*this); ret.limitTo(o): return ret; }
```
?

Reply:
∩ is called intersection. So let's be consistent with your suggestion to ```normalized```, and rename intersect to intersection and limitTo to intersect. 
While at it, I'll add some constexpr, which sadly doesn't allow for you second suggestion until C++14.